src/string_radix_sort.jl

```diff
@@ -0,0 +1,233 @@
+import Base: Forward, ForwardOrdering, Reverse, ReverseOrdering, Lexicographic, LexicographicOrdering, sortperm, Ordering, 
+            setindex!, getindex, similar
+import SortingAlgorithms: RadixSort, RadixSortAlg
+"""
+    load_bits([type,] s, skipbytes)
+
+Load the underlying bits of a string `s` into a `type` of the user's choosing.
+The default is `UInt`, so on a 64 bit machine it loads 64 bits (8 bytes) at a time.
+If the `String` is shorter than 8 bytes then it's padded with 0. 
+
+Assumes machine is little-endian
+
+- `type`:       any bits type that has `>>`, `<<`, and `&` operations defined
+- `s`:          a `String`
+- `skipbytes`:  how many bytes to skip e.g. load_bits("abc", 1) will load "bc" as bits
+"""
+load_bits(s::String, skipbytes = 0) = load_bits(UInt, s, skipbytes)
+
+function load_bits(::Type{T}, s::String, skipbytes = 0) where T
+    n = sizeof(s)
+    if n < skipbytes
+        return zero(T)
+    elseif n - skipbytes >= sizeof(T)
+        return ntoh(unsafe_load(Ptr{T}(pointer(s, skipbytes+1))))
+    else
+        ns = (sizeof(T) - min(sizeof(T), n - skipbytes))*8
+        h = unsafe_load(Ptr{T}(pointer(s, skipbytes+1)))
```

Comment:
Couldn't this segfault if you read past the end of the string?   e.g. if the string ends on a page boundary and the next page is inaccessible?

Reply:
made a segfault safe version